### PR TITLE
Split out S3 backend for Terraform

### PR DESCRIPTION
**Pull request by codetheweb**
A few changes I needed to make to get started.

I split out the S3 backend from the main deployment as it seems like the recommendation is to deploy it once manually rather than continously to avoid the chicken and egg problem.

Also, the Terraform backend config is now in a separate file to make it easier to use a local dev environment.

### CODE REVIEW

File: terraform/compass_module/main.tf [hunk 1]
```diff
@@ -15,46 +11,4 @@ terraform {
 
 provider "aws" {
   region  = var.aws_region
-  profile = var.aws_user
```
Comment (codetheweb) [inline terraform/compass_module/main.tf:18]: needed to remove this to be able to run, looks like there's other ways to set this (e.x. through an environment variable)

File: terraform/s3-backend/.terraform.lock.hcl
Comment (codetheweb) [inline terraform/s3-backend/.terraform.lock.hcl:1]: terraform recommends that lock files be checked in

File: terraform/staging/main.tf [hunk 2]
```diff
@@ -1,10 +1,6 @@
 terraform {
   backend "s3" {
-    bucket         = "compass-staging-tf-state" # Substitute environment as needed
-    key            = "terraform.tfstate"
-    region         = "us-west-2"
-    dynamodb_table = "compass-staging-tf-state-locking" # Substitute environment as needed
-    encrypt        = true
+    key = "staging/terraform.tfstate"
```
Comment (codetheweb) [inline terraform/staging/main.tf:3]: recommended to use the same state bucket for multiple environments

**Review by amantri — APPROVED** [review]

**Comment by codetheweb** [comment]
Now using GCS, so closing.

**Pull request closed without merging**